readthedocs/embed/v3/views.py

```diff
@@ -163,32 +172,26 @@ def _parse_based_on_doctool(self, page_content, fragment, doctool, doctoolversio
                     # ...
                     # </dl>
 
-                    # TODO: figure it out if it's needed to remove the siblings here
-                    # parent = node.parent
-                    # for n in parent.traverse():
-                    #     if n not in (node, node.next):
-                    #         n.remove()
-                    node = node.parent
-
-                elif 'citation' in node.parent.attributes.get('class'):
-                    # Sphinx HTML structure for sphinxcontrib-bibtex puts the ``id`` in the
-                    # ``dt`` element with the title of the cite. In this case, we
-                    # return the parent node which contains the definition list
-                    # and remove all ``dt/dd`` that are not the requested one
-
-                    # Structure:
-                    # <dl class="citation">
-                    # <dt id="cite-id"><span><a>Title of the cite</a></span></dt>
-                    # <dd>Content of the cite</dd>
-                    # ...
-                    # </dl>
-
-                    # TODO: figure it out if it's needed to remove the siblings here
-                    # parent = node.parent
-                    # for n in parent.traverse():
-                    #     if n not in (node, node.next):
-                    #         n.remove()
-                    node = node.parent
+                    parent_node = node.parent
+                    if 'glossary' in node.parent.attributes.get('class'):
+                        next_node = node.next
+
+                    elif 'citation' in node.parent.attributes.get('class'):
+                        next_node = node.next.next
+
+                    # Iterate over all the siblings (``.iter()``) of the parent
+                    # node and remove ``dt`` and ``dd`` that are not the ones
+                    # we are looking for. Then return the parent node as
+                    # result.
+                    #
+                    # Note that ``.iter()`` returns a generator and we modify
+                    # the HTML in-place, so we have to convert it to a list
+                    # before removing elements. Otherwise we break the
+                    # iteration before compliting it
```

Comment:
```suggestion
                    # Note that ``.iter()`` returns a generator and we modify
                    # the HTML in-place, so we have to convert it to a list
                    # before removing elements. Otherwise we break the
                    # iteration before completing it
```